### ISSUE TITLE
argument-list-wrapping should ignore comments at the end of the line

### ISSUE THREAD

**Issue by mklnwt** [pinterest/ktlint#2450]
```Kotlin
// Assume that the last allowed character is at the X character on the right             X
val foo = Bar.builder().baz(0).build() // long comment at the end of the line regarding baz
```

## Expected Behavior
Rule should ignore the comment. This should rather be caught by `max-line-length`.

The comment might relate to something within the line and could be auto-formatted out of context.

## Current Behavior
```Kotlin
val foo = Bar.builder().baz(
    0,
).build() // long comment at the end of the line regarding baz
```

## Additional information
* Current version of ktlint: 1.1.0

**Comment by mklnwt** [comment]
Another example:

```Kotlin
// Assume that the last allowed character is
// at the X character on the right                                  X
@Foo(bar = "baz") // long comment at the end of the line regarding baz
```

produces

```Kotlin
@Foo(
    bar = "baz",
) // long comment at the end of the line regarding baz
```

**Comment by paul-dingemans** [comment]
> @Foo(
>     bar = "baz",
> ) // long comment at the end of the line regarding baz
> 

Yeah, this is not great.  I guess that the comment explains why the annotation is used. I find it questionable whether EOL comments should be used for that purpose. I would expect that the `@Foo` name should express that sufficiently.

Regardless of the content of the comment, I think it is important that when a comment is written that it is clearly visible to the developer. Consider that your screen would be as wide that character `X` in examples below is just visible. In the first example, the start of the comment is visible and as of that will invite the reader to scroll to the right in case the comment stops at an unexpected place. In the second example, the comment just starts outside of the visible window, which I would consider bad.
```
// Assume that the last allowed character is
// at the X character on the right                                  X
val foo = "fff" // long long long long long long long long long long long long long long long long long long long long long long long long long long long

@Foo(bar = "bazzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz") // long comment at the end of the line regarding baz
```

Note that EOL comments that are placed on a separate line are already allowed to exceed the maximum line length.
```
// long long long long long long long long long long long long long long long long long long long long long long long long long long long long
```

If EOL comments are to be ignored, a number of rules have to be changed for a consistent experience. Handling comments is already painful enough. Adding this exception would complicate multiple rules, possible at the cost of

**Comment by mklnwt** [comment]
I agree that the comments should not be extending as far out.

> If EOL comments are to be ignored, a number of rules have to be changed for a consistent experience. Handling comments is already painful enough. Adding this exception would complicate multiple rules.

This seems like the right thing to do though. Marking EOL comments as exceeding `max-line-length` and not auto-correcting it so the developer can decide where to place the comment best. Similar to `value-argument-comment` & `value-parameter-comment`.

**Comment by paul-dingemans** [comment]
Ktlint does not distinct 'warnings' and 'errors'. So either a comment exceeding `max-line-length` is reported and must be fixed by the developer. Or, a comment exceeding `max-line-length` is not reported at all. 

I will think about alternatives.

**Comment by mklnwt** [comment]
I do mean it should be reported and fixed by the developer.
It should just not be auto-formatted.
That's why the other rules should ignore EOL comments and let `max-line-length` catch it.

**Comment by paul-dingemans** [comment]
@mklnwt I have checked all rules for the way they use `max_line_length` and fixed a couple of them. In other rules, I was not able to create examples in which a EOL-comment would cause an unwanted violation. I would appreciate if you can test latest [SNAPSHOT build](https://pinterest.github.io/ktlint/latest/install/snapshot-build/) to check for issues regarding EOL-comments.

**Comment by mklnwt** [comment]
@paul-dingemans I checked and it seems to have done the trick!

Thanks again!
